projects/extension/src/content/ExtensionMessageRouter.ts

```diff
@@ -101,56 +102,28 @@ export class ExtensionMessageRouter {
     port.postMessage(msg)
   }
 
-  #disconnectPort = ({ chainName, chainId }: ToExtension): void => {
-    const port = this.#ports[chainId]
-
-    if (!port) {
-      // probably someone trying to abuse the extension.
-      console.warn(`App requested to disconnect ${chainName} - no port found`)
-      return
-    }
-
-    port.disconnect()
-    debug(`DISCONNECTED ${chainName} PORT`, port)
-    delete this.#ports[chainId]
-    return
-  }
-
   #handleMessage = (msg: MessageEvent<ToExtension>): void => {
     const data = msg.data
-    const { origin, action, type } = data
+    const { origin, type } = data
     if (!origin || origin !== EXTENSION_PROVIDER_ORIGIN) {
       return
     }
 
     debug(`RECEIVED MESSAGE FROM ${EXTENSION_PROVIDER_ORIGIN}`, data)
 
-    if (!action) {
-      return console.warn("Malformed message - missing action", msg)
-    }
-
-    if (action === "disconnect") {
-      return this.#disconnectPort(data)
+    if (!type) {
+      // probably someone abusing the extension
+      console.warn("Malformed message - missing message.type", data)
+      return
     }
```

Comment:
If `type` is no longer optional we can remove this, no?

Reply:
I guess it's still relevant in case that the consumer is not using TS.

Reply:
TS is supposed to check whether this is valid. If we can't rely on TS's checks, then there's a ton of things to verify.

Reply:
what I meant is that it's possible that someone writes a client in plain JS that interacts with the Extension... TS doesn't do anything on the runtime.

Reply:
So when the content script does this: https://github.com/paritytech/substrate-connect/blob/73d0e96ae46d7b309eb5c73a8aa2bfdbde90c417/projects/extension/src/content/ExtensionMessageRouter.ts#L43

And someone does `window.postMessage({ "some_garbage": 5 })`, then `handleMessage` will be called with `{ "some_garbage": 5 }` passed as parameter and throw exceptions?